src/Components/CreateImageWizardV2/steps/Review/ReviewStepTextLists.tsx

```diff
@@ -107,15 +108,21 @@ export const FSCList = () => {
 };
 
 export const TargetEnvAWSList = () => {
-  const { data: rawAWSSources, isSuccess } = useGetSourceListQuery({
+  const { isSuccess } = useGetSourceListQuery({
     provider: 'aws',
   });
-  const awsAccountId = useAppSelector((state) => selectAwsAccountId(state));
+  const awsAccountID = useAppSelector((state) => selectAwsAccountId(state));
   const awsShareMethod = useAppSelector((state) => selectAwsShareMethod(state));
+  const source = useAppSelector((state) => selectAwsSource(state));
 
-  const awsSources = extractProvisioningList(rawAWSSources);
-  const { isBeta } = useGetEnvironment();
+  const { data } = useGetSourceUploadInfoQuery(
+    {
+      id: parseInt(source?.id as string),
+    },
+    { skip: source === undefined }
+  );
 
+  const getAccountId = data?.aws?.account_id;
```

Comment:
This is all a bit confusing, and it is entirely my fault. :sweat_smile: 

First, a quick side note... I don't like this variable name. When I see the `get` prefix, I assume this is a getter function... but it is actually a string, not a function.

But the more confusing thing here is that we can't just use the value of `awsAccountId` because in the case that we're using a source, its value is `''`. 

I think that is what we should actually fix.

Would you mind updating this so that on the AWS step we set the value of `awsAccountId` when we are using sources? Then we won't need to call the `useGetSourceUploadInfoQuery` hook at all, we can just use `awsAccountId` for both cases.